src/explore-education-statistics-frontend/src/middleware/pages/redirectPages.ts

```diff
@@ -0,0 +1,62 @@
+import redirectService, {
+  Redirects,
+  RedirectType,
+} from '@frontend/services/redirectService';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+interface StoredRedirects {
+  redirects: Redirects;
+  fetchedAt: number;
+}
+
+let storedRedirects: StoredRedirects | undefined;
+
+const redirectPaths = {
+  methodologyRedirects: '/methodology',
+  publicationRedirects: '/find-statistics',
+};
+
+const cacheTime = 300000;
+
+export default async function redirectPages(request: NextRequest) {
+  const shouldRefetch =
+    !storedRedirects ||
+    (storedRedirects?.fetchedAt &&
+      storedRedirects.fetchedAt + cacheTime < Date.now());
+
+  if (shouldRefetch) {
+    storedRedirects = {
+      redirects: await redirectService.list(),
+      fetchedAt: Date.now(),
+    };
+  }
+
+  let redirectUrl: string | undefined;
```

Comment:
Rather than neding this `let`, could you instead just replace the assignment of `redirectUrl` with:

```
const redirectUrl = pathNameSections
          .map(section =>
            section === rewriteRule?.fromSlug ? rewriteRule?.toSlug : section,
          )
          .join('/');
return NextResponse.redirect(new URL(redirectUrl, request.url))
```

No biggie either way, but I've always had it drummed into me to try to avoid lets where possible!

Reply:
That didn't quire work but I've refactored it to use a reduce instead so that a `let` isn't needed.

Reply:
Haha appreciated - hopefully worth the effort :)